src/unix/wayland.zig

```diff
@@ -400,6 +401,16 @@ pub fn setCursorMode(self: *@This(), mode: wio.CursorMode) void {
     }
 }
 
+pub fn setSize(self: *@This(), size: wio.Size) void {
+    self.resizing = true;
+    self.size = size;
+    c.libdecor_frame_unset_fullscreen(self.frame);
```

Comment:
I think this should be the other way around: `frameConfigure` calls `setSize` (and move the resize logic there).